core/src/history.rs

```diff
@@ -32,6 +32,11 @@ impl ExchangeHistory {
         ExchangeHistory::read(File::open(filestore)?)
     }
 
+    /// Returns the very first batch for the exchange.
+    pub fn first_batch(&self) -> Option<BatchId> {
+        self.events.events().next().map(|(_, batch)| batch)
```

Comment:
Does this work on the assumption that the first event emitted contains a batch information element? If not, how does this get the first batch?